### PR TITLE
Tweak the editor experience for event series. DDFFORM-125 

### PR DESCRIPTION
**Pull request by rasben**
Overall, this is an admin experience thing, and is best tested by going into the environment and logging in as an `editor` user.
I've added the editor user in 1password

---------

**Tweak the editor experience for event series. DDFFORM-125**

- Hide/disable the existing 'views' from recurring_events module,
  as they are too confusing.
- Create our own admin view for events overview, including some
  custom module functionality for admin clarity.
- Add some admin styling, that is also general enough for non-events.

-----

**Add exisiting teaser image/text to event instances.**

The fields already existed on the node type, but not on the
event instance entity types. These fields are used for
card displays.

-----

**Streamline and simplify events templates.**

This commit removes a bunch of redundancy markup in the
event template files, instead replacing it with simple loops
that are easier to maintain.
Also updates the include logic between event nodes and event instances.

------

**Add test Drupal users, when running dev / PR site resets.**

Using the same password as admin users - will update 1password.

------

**Disallow users to add event nodes, as they are deprecated.**

They have been replaced with event_series / event_instances,
and this CT will completely be removed in the future.

### CODE REVIEW

**Review by kasperg — CHANGES_REQUESTED** [review]
:+1: Lots of code, lots of comments. Good stuff!

File: config/sync/field.field.eventinstance.default.field_teaser_image.yml
[the remark this answers is not in the document]
Comment (rasben) [reply config/sync/field.field.eventinstance.default.field_teaser_image.yml:1]: 👍 I didnt know i had to do it manually

File: Taskfile.yml [hunk 1]
```diff
@@ -181,6 +183,24 @@ tasks:
...
+      - task dev:cli -- drush user:role:add 'administrator' administrator
+
+      - task dev:cli -- drush user:create mediator --password="test"
+      - task dev:cli -- drush user:role:add 'mediator' mediator
+
+      - task dev:cli -- drush user:create local_administrator --password="test"
+      - task dev:cli -- drush user:role:add 'local_administrator' local_administrator
+
+      - task dev:cli -- drush user:create patron --password="test"
+      - task dev:cli -- drush user:role:add 'patron' patron
```
[the remark this answers is not in the document]
Comment (rasben) [reply Taskfile.yml:202]: oops, that was the idea, but i forgot it : ' )

File: Taskfile.yml [hunk 2]
```diff
@@ -80,6 +80,8 @@ tasks:
       - task dev:cli -- php vendor/bin/phpcs
       - task dev:cli -- php vendor/bin/twig-cs-fixer lint --fix
       - task dev:cli -- php vendor/bin/phpstan
+      - yarn install --frozen-lockfile
+      - npx eslint web -c .eslintrc.json --fix
```
[the remark this answers is not in the document]
Comment (rasben) [reply Taskfile.yml:84]: I'd love to add a explanation, but I have no idea why. I'm just following the precedent.

From what I can see, the other npm commands in the taskfile.yml are also run locally, and there isnt even a NPM docker container I can run it in

File: web/modules/custom/dpl_admin/dpl_admin.module [hunk 3]
```diff
@@ -19,3 +26,147 @@ function dpl_admin_preprocess_html(array &$variables): void {
...
+  $output = $variables['output'];
+
+  if (!$view instanceof ViewExecutable || $view->id() !== 'event_admin') {
+    return;
+  }
+
+  if (!$field instanceof EntityField ||
+    !$output instanceof Markup ||
+    $field->getField() !== '.eventseries_id' ||
+    !method_exists($output, '__toString')) {
```
[the remark this answers is not in the document]
Comment (rasben) [reply web/modules/custom/dpl_admin/dpl_admin.module:68]: You'd think so, but PHPStan is very upset if i dont add this line.
I think it's a bug with PHPStan

File: web/modules/custom/dpl_admin/dpl_admin.module [hunk 4]
```diff
@@ -19,3 +26,147 @@ function dpl_admin_preprocess_html(array &$variables): void {
...
+ */
+function _dpl_admin_form_alter_eventinstance(array &$form, FormStateInterface $form_state, string $form_id): void {
+  $event_instance = \Drupal::routeMatch()->getParameter('eventinstance');
+  $event_series_edit_url = NULL;
+
+  if ($event_instance instanceof EventInstance) {
+    $event_series = $event_instance->getEventSeries();
+
+    /** @var \Drupal\recurring_events\EventInterface $event_series */
+    if (method_exists($event_series, 'id')) {
```
[the remark this answers is not in the document]
Comment (rasben) [reply web/modules/custom/dpl_admin/dpl_admin.module:139]: I ran into the same PHPStan issue as I mentioned before.

File: config/sync/field.field.eventseries.default.field_teaser_image.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  config:
+    - field.storage.eventseries.field_teaser_image
+    - media.type.image
+    - recurring_events.eventseries_type.default
+id: eventseries.default.field_teaser_image
+field_name: field_teaser_image
+entity_type: eventseries
+bundle: default
+label: 'Teaser image'
+description: "The teaser fields are used for the card of display.\r\nIf no image has been selected, the text will be shown instead:\r\n\r\n<img src=\"/themes/custom/novel/images/teaser-text-image.jpg\" />\r\n\r\n<hr/>"
```
[the remark this answers is not in the document]
Comment (rasben) [reply config/sync/field.field.eventseries.default.field_teaser_image.yml:14]: I think it's a good point, but, as you also allude to, it's outside the scope of events, so i think it's a conversation we should have with DDF during a demo

File: config/sync/views.view.event_admin.yml
[the remark this answers is not in the document]
Comment (rasben) [reply config/sync/views.view.event_admin.yml:1]: I'll add a 'edit series' button, but further discussion is something i think should happen during demo, as i'm kinda winging it here :)

File: web/modules/custom/dpl_admin/dpl_admin.module (hunk 3 above)
[the remark this answers is not in the document]
Comment (kasperg) [reply web/modules/custom/dpl_admin/dpl_admin.module]: That was not the case. I could remove it without problems in fa35d7a1fe430f2b836696baec745c26a89974c9.

File: web/modules/custom/dpl_admin/dpl_admin.module (hunk 4 above)
[the remark this answers is not in the document]
Comment (kasperg) [reply web/modules/custom/dpl_admin/dpl_admin.module]: This turned out to be a problem with the Recurring Events module: https://www.drupal.org/project/recurring_events/issues/3415222.

I have fixed this in 795176a542acb739a3adbafb78d3770bf48a2d3a.

File: config/sync/field.field.eventinstance.default.field_teaser_image.yml
[the remark this answers is not in the document]
Comment (kasperg) [reply config/sync/field.field.eventinstance.default.field_teaser_image.yml:1]: I have done so in 8c0891eda540a6a707f10d9fb9592fd2fd0d92b7 and c8104ddfc98ad012a6d951e786f985cfd99170a5.

File: config/sync/recurring_events.eventseries.config.yml [hunk 6]
```diff
@@ -9,7 +9,7 @@ days: 'monday,tuesday,wednesday,thursday,friday,saturday,sunday'
 limit: 10
 excludes: 1
 includes: 1
-enabled_fields: 'consecutive_recurring_date,daily_recurring_date,weekly_recurring_date,monthly_recurring_date,yearly_recurring_date,custom'
```
[the remark this answers is not in the document]
Comment (kasperg) [reply config/sync/recurring_events.eventseries.config.yml:12]: I have reenabled it in 892f3b2ed8f725e00e3fa50d62d6804837453efc.

**Pull request merged**